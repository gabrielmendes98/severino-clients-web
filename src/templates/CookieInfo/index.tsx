import Link from 'next/link';
import NoSsr from '@mui/material/NoSsr';
import useMediaQuery from '@mui/material/useMediaQuery';
import { useDispatch, useSelector } from 'common/store/hooks';
import { cookieConsent, selectCookieConsented } from 'common/slices/cookie';
import useAppTheme from 'common/hooks/useTheme';
import Paper from 'components/Paper';
import Text from 'components/Text';
import Grid from 'components/Grid';
import Button from 'components/Button';

const CookieInfo = () => {
  const theme = useAppTheme();
  const cookieConsented = useSelector(selectCookieConsented);
  const dispatch = useDispatch();
  const isSmallScreen = useMediaQuery(theme.breakpoints.down('md'));

  const handleAccept = () => {
    dispatch(cookieConsent());
  };

  if (cookieConsented) {
    return null;
  }

  return (
    <NoSsr>
      <Paper
        sx={{
          position: 'fixed',
          width: '80%',
          bottom: theme => theme.spacing(3),
          left: 0,
          right: 0,
          margin: '0 auto',
          bgcolor: 'background.paper',
          p: 2,
        }}
      >
        <Grid container rowSpacing={2}>
          <Grid container item xs={12} md={11} flexDirection="column">
            <Text variant="h6">Nós respeitamos sua privacidade</Text>
            <Text>
              Utilizamos cookies para salvar suas credenciais, sua localização e
              o tema da aplicação. Entenda como funciona em nossa{' '}
              <Link href="/cookie-policy">política de cookies</Link>
            </Text>
          </Grid>

          <Grid
            container
            item
            xs={12}
            md={1}
            alignItems="center"
            justifyContent="flex-end"
          >
            <Button
              variant="contained"
              fullWidth={isSmallScreen}
              onClick={handleAccept}
            >
              Entendi
            </Button>
          </Grid>
        </Grid>
      </Paper>
    </NoSsr>
  );
};

export default CookieInfo;
