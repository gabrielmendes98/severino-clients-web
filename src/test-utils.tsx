import React, { FC, ReactElement, ReactNode } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { PreloadedState } from '@reduxjs/toolkit';
import ThemeProvider from './common/providers/Theme';
import { AppStore, RootState, setupStore } from './common/store/store';

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>;
  store?: AppStore;
}

const customRender = (
  ui: ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = setupStore(preloadedState),
    ...renderOptions
  }: ExtendedRenderOptions = {},
) => {
  const Wrapper: FC<{ children: ReactNode }> = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider>{children}</ThemeProvider>
    </Provider>
  );

  return { ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
};

/* eslint-disable */
export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
/* eslint-enable */
