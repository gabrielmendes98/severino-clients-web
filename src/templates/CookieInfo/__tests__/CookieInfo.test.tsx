import { render } from 'test-utils';
import CookieInfo from '../index';

it('should display nothing if cookie is consented', () => {
  const { container } = render(<CookieInfo />, {
    preloadedState: {
      cookie: {
        accepted: true,
      },
    },
  });

  expect(container).toMatchSnapshot();
});

it('should call api and set cookie consent to true when accept', () => {});
