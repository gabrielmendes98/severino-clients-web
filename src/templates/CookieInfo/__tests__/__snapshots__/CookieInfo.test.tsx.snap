// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display nothing if cookie is consented 1`] = `
<div>
  <div
    class="Toastify"
  />
</div>
`;
